doc/content/toolstack/features/NUMA/index.md

```diff
@@ -62,39 +62,59 @@ The Xen scheduler supports 2 kinds of constraints:
+As a result, restarting processes will speed them up or slow them down as address space randomization picks different memory regions inside a VM.
+
+This uses the memory bandwidth of all memory controllers and distributes the load across all nodes.
+However, the memory latency is higher as the NUMA interconnects are used for most memory accesses and vCPU synchronization within the Domains.
 
 Note that this is not the worst case: the worst case would be for memory to be allocated on one NUMA node, but the vCPU always running on the furthest away NUMA node.
 
 ## Best effort NUMA-aware memory allocation for VMs
 
-By default Xen stripes the VM's memory accross all NUMA nodes of the host, which means that every VM has to go through all the interconnects. 
+
+### Summary
+
+The best-effort mode attempts to fit Domains into NUMA nodes and to balance memory usage.
+It soft-pins Domains on the NUMA node with the most available memory when adding the Domain.
+Memory is currently allocated when booting the VM (or while constructing the resuming VM).
+
+Parallel boot issue: Memory is not pre-allocated on creation, but allocated during boot.
+The result is that parallel VM creation and boot can exhaust the memory of NUMA nodes.
+
+### Goals
+
+By default, Xen stripes the VM's memory across all NUMA nodes of the host, which means that every VM has to go through all the interconnects.
 The goal here is to find a better allocation than the default, not necessarily an optimal allocation.
-An optimal allocation would require knowing what VMs you would start/create in the future, and planning across hosts too.
+An optimal allocation would require knowing what VMs you would start/create in the future, and planning across hosts.
+This allows the host to use all NUMA nodes to take advantage of the full memory bandwidth available on the pool hosts.
 
-Overall we want to balance the VMs across NUMA nodes, such that we use all NUMA nodes to take advantage of the maximum memory bandwidth available on the system.
+Overall, we want to balance the VMs across NUMA nodes, such that we use all NUMA nodes to take advantage of the maximum memory bandwidth available on the system.
 For now this proposed balancing will be done only by balancing memory usage: always heuristically allocating VMs on the NUMA node that has the most available memory.
-Note that this allocation has a race condition for now when multiple VMs are booted in parallel, because we don't wait until Xen has constructed the domain for each one (that'd serialize domain construction, which is currently parallel). 
+For now, this allocation has a race condition: This happens when multiple VMs are booted in parallel, because we don't wait until Xen has constructed the domain for each one (that'd serialize domain construction, which is currently parallel).
```

Comment:
I have the same issue with regards to this statement as above: vcpu pinning is serialized and take sinto account the memory reservation that are currently happening. If vcpu pinning is taken into account when performing the memory reservation, this race condition should not be present